src/app/users/associateRoles.js

```diff
@@ -15,6 +15,13 @@ const renderAssociateRolesPage = (req, res, model) => {
 };
 
 const buildBackLink = (req, currentServiceIndex) => {
+  const isRequestServiceUrl = isRequestService(req)
+  if(isRequestServiceUrl) {
+    const sid = req.session.user.serviceId ? req.session.user.serviceId : ""
```

Comment:
what would happen if service is not defined and we pass an empty string?

Reply:
It shouldn't happen as I am setting this in the user session when the `approveServiceRequest`  page load, but as an extra check I just added this in the `if` condition.